runtime/vm/ContinuationHelpers.cpp

```diff
@@ -167,10 +168,17 @@ synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuati
 	 * but only up to small finite number of times.
 	 */
 	do {
+		if (oldContinuationState == returnContinuationState) {
+			oldContinuationState = newContinuationState;
+
+		} else {
+			oldContinuationState = returnContinuationState;
+		}
```

Comment:
this is hard to follow
either we need more comments or better we just start this loop from scratch: read oldContinuationState from continuationStatePtr and check isConcurrentlyScanned against oldContinuationState...